drivers/hmis/app/graphql/types/hmis_schema/assessment_input.rb

```diff
@@ -34,6 +34,14 @@ def find_or_create_assessment
 
       return [nil, errors.errors] if errors.any?
 
+      # Check if it's valid to create a new assessment of this type
+      unless assessment.present?
+        errors.add :assessment, :invalid, full_message: 'An intake assessment for this enrollment already exists.' if form_definition.intake? && enrollment.intake_assessment.present?
+        errors.add :assessment, :invalid, full_message: 'An exit assessment for this enrollment already exists.' if form_definition.exit? && enrollment.exit_assessment.present?
```

Comment:
Not sure this is what you're asking for, but it would be DRYed up a bit by making a class method on Enrollment or Exit that returns the error message (perhaps given a symbol, or just individual methods per message.